src/floorist/floorist.py

```diff
@@ -52,7 +53,18 @@ def main():
                     date.today().strftime('year_created=%Y/month_created=%-m/day_created=%-d')
                 ])
 
+                uuids = {}
+
                 for data in cursor:
+                    if len(uuids) == 0 and len(data) > 0:
+                        # Detect any columns with UUID
+                        for column in data:
+                            if type(data[column][0]) == uuid.UUID:
+                                uuids[column] = "string"
+
+                    # Convert any columns with UUID type to string
+                    data = data.astype(uuids)
+
```

Comment:
I don't understand why we need  `uuids` ... I'd like also to unit test this function, so if we can extract this functionality to a method. I think this should work:

```python
def convert_UUID_columns_to_string(data):

    for column in data:
        for value in data[column]:
            if isinstance(value, UUID):
                data[column] = data[column].astype("string")
                continue
# ...
                for data in cursor:
                    convert_UUID_columns_to_string(data)
# ...
```

Reply:
Performance, we do this in batches and only want to check the columns for the first batch.

Reply:
oh LOL - I meant `break` not `continue`.

```python
def convert_UUID_columns_to_string(data):
 
    for column in data:
        for value in data[column]:
            if isinstance(value, uuid.UUID):
                data[column] = data[column].astype("string")
                break

    return data
# ...
    for data in cursor:
         data = convert_UUID_columns_to_string(data)
# ...

```

I don't understand the mention to do this in batches, this is meant to be inside the `for data in cursor` body, I mean, inside the batches loop, so it behaves as you'd expect (treating each of the batches one at a time)

Also, the reason I'm iterating over values in the data column is because I don't assume the field will have a value. This is of course, slower, as I'm iterating over the values to find one that's of the type  

I've measured performance with `2000000` records (per batch) these are the results:

```
# my suggested implementation
took 2.5525994300842285 seconds

# The PR's original implementation
took 1.8659965991973877 seconds

# my suggested implementation just checking the first value as the PR does
took 1.9471654891967773 seconds
```

If you assume there's always going to be a value on the first value of each column, then either implementation is about the same level of performance. 

If you want to go with yours, I'm okay, but please at least:
- extract that logic to a method so it's easily testable
- use `isinstance` instead of `type` - usually type is reserved for debug purposes.

Reply:
I do understand why you want this to be testable, but the code is still simple enough for not having unit tests and I would like to keep it that way as long as possible. The reason for this is that we're planning a rewrite of floorist into an operator. If you look at the current tests, they are all just running the `main` function and the reason behind this is to be able to use the same exact tests even if the codebase gets rewritten to go.